app/services/foreman_bootdisk/iso_generator.rb

```diff
@@ -73,17 +93,35 @@ def self.generate(opts = {}, &block)
         end if opts[:files].respond_to? :each
       end
 
+      if opts[:grub]
+        FileUtils.mkdir_p(File.join(wd, 'build', 'EFI', 'BOOT'))
+        FileUtils.cp('/var/lib/tftpboot/grub2/shimx64.efi', File.join(wd, 'build', 'EFI', 'BOOT', 'BOOTX64.efi'))
```

Comment:
We cannot use this path directly, it won't be present on Foreman when Smart Proxy TFTP feature was not installed. Can you add new `Setting[:bootdisk_grub2_dir]` which will default to `/boot/efi/EFI/*` and use that here and on the other three lines? You will need to expand shell glob because the path is different on different platforms: /boot/efi/EFI/fedora vs /boot/efi/EFI/redhat vs /boot/efi/EFI/centos etc.

Reply:
unfortunately the grubx64.efi provided in that directory doesn't include the linux and initrd commands... needs change of grub2 rpm in os to work.

Reply:
from grub2.spec in RHEL7

```
GRUB_MODULES="  all_video boot btrfs cat chain configfile echo efifwsetup \
                efinet ext2 fat font gfxmenu gfxterm gzio halt hfsplus iso9660 \
                jpeg linuxefi lvm minicmd normal part_apple part_msdos \
                part_gpt password_pbkdf2 png reboot search search_fs_uuid \
                search_fs_file search_label sleep test video xfs \
                mdraid09 mdraid1x blscfg multiboot2 multiboot tftp"
```

from foreman-generate-bootloaders cmd:
```
MODULES="all_video boot btrfs cat configfile echo ext2 fat font gfxmenu gfxterm gzio halt hfsplus iso9660 jpeg loadenv loopback lvm mdraid09 mdraid1x minicmd normal part_apple part_msdos part_gpt password_pbkdf2 png reboot search search_fs_uuid search_fs_file search_label serial sleep syslinuxcfg test tftp video xfs linux"
```

Reply:
Good point, can you make a comment in the setting "Install foreman-bootloaders-redhat package or copy shimx64.efi and grubx64.efi manually" so users can figure this out both on CentOS/RHEL and Debian/Ubuntu? We don't generate those on Debians.

Reply:
And of course keep using the `/var/lib/tftpboot/grub2` path, that's fine.